src/frontend/apps/desk/src/features/mail-domains/components/useAddMailDomainApiError.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { UseFormReturn } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+
+import { APIError } from '@/api';
+
+export const useAddMailDomainApiError = ({
+  error,
+  methods,
+}: {
+  error: APIError | null;
+  methods: UseFormReturn<{ name: string }> | null;
+}): string[] | undefined => {
+  const [errorCauses, setErrorCauses] = React.useState<string[]>([]);
+  const { t } = useTranslation();
+
+  React.useEffect(() => {
+    if (methods && t && error) {
+      let causes: undefined | string[] = undefined;
+
+      if (error.cause?.length) {
+        const parseCauses = (causes: string[]) =>
+          causes.reduce((arrayCauses, cause) => {
+            switch (cause) {
+              case 'Mail domain with this name already exists.':
+              case 'Mail domain with this Slug already exists.':
+                methods.setError('name', {
+                  type: 'manual',
+                  message: t(
+                    'This mail domain is already used. Please, choose another one.',
+                  ),
+                });
+                methods.setFocus('name');
+                break;
+              default:
+                arrayCauses.push(cause);
+            }
+
+            return arrayCauses;
+          }, [] as string[]);
+
+        causes = parseCauses(error.cause);
+      }
+
+      if (error.status === 500 || !error.status) {
+        const error500Message = t(
+          'Your request cannot be processed because the server is experiencing an error. If the problem ' +
+            'persists, please contact our support to resolve the issue: suiteterritoriale@anct.gouv.fr',
+        );
+        causes =
+          causes && causes?.length
+            ? [error500Message, ...causes]
+            : [error500Message];
+      }
+
+      setErrorCauses((stateCauses) =>
+        causes && JSON.stringify(causes) !== JSON.stringify(stateCauses)
+          ? causes
+          : stateCauses,
+      );
+    }
+  }, [methods, t, error]);
+
+  React.useEffect(() => {
+    if (errorCauses?.length && methods) {
+      methods.setFocus('name');
+    }
+  }, [methods, errorCauses]);
```

Comment:
I have the feeling this useEffect is not necessary, could you add the setFocus somewhere in the first useEffect ?